### PR TITLE
[3.7.2] GooglePlayServices availability check.

### PR DESCRIPTION
**Pull request by paynd**

### CODE REVIEW

File: android/src/main/java/com/gettipsi/stripe/StripeModule.java [hunk 1]
```diff
@@ -135,6 +136,10 @@ public void init(ReadableMap options) {
 
   @ReactMethod
   public void deviceSupportsAndroidPay(final Promise promise) {
+    if(!isPlayServicesAvailable()){
+      promise.reject(TAG, "Play services not available!");
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/StripeModule.java]:
```
Play services are not available!
```

File: android/src/main/java/com/gettipsi/stripe/StripeModule.java [hunk 2]
```diff
@@ -308,6 +313,15 @@ private MaskedWalletRequest createWalletRequest(final String estimatedTotalPrice
     return maskedWalletRequest;
   }
 
+  private boolean isPlayServicesAvailable() {
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/StripeModule.java]:
```
arePlayServicesAvailable
```

File: android/src/main/java/com/gettipsi/stripe/StripeModule.java [hunk 3]
```diff
@@ -135,6 +136,10 @@ public void init(ReadableMap options) {
 
   @ReactMethod
   public void deviceSupportsAndroidPay(final Promise promise) {
+    if(!isPlayServicesAvailable()){
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/StripeModule.java]:
```
if (!arePlayServicesAvailable()) {
```

File: android/src/main/java/com/gettipsi/stripe/StripeModule.java [hunk 4]
```diff
@@ -308,6 +313,15 @@ private MaskedWalletRequest createWalletRequest(final String estimatedTotalPrice
     return maskedWalletRequest;
   }
 
+  private boolean isPlayServicesAvailable() {
+    GoogleApiAvailability googleAPI = GoogleApiAvailability.getInstance();
+    int result = googleAPI.isGooglePlayServicesAvailable(getCurrentActivity());
+    if(result != ConnectionResult.SUCCESS) {
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/StripeModule.java]:
```
if (result != ConnectionResult.SUCCESS) {
```

File: android/src/main/java/com/gettipsi/stripe/StripeModule.java [hunk 5]
```diff
@@ -135,6 +136,10 @@ public void init(ReadableMap options) {
 
   @ReactMethod
   public void deviceSupportsAndroidPay(final Promise promise) {
+    if (!isPlayServicesAvailable()){
```
Comment (isnifer) [inline android/src/main/java/com/gettipsi/stripe/StripeModule.java]:
```diff
- if (!isPlayServicesAvailable()){
+ if (!isPlayServicesAvailable()) {
```

**Comment by cybergrind** [comment]
@isnifer is it should be merged before #120 ?

**Comment by isnifer** [comment]
@cybergrind yes, sure)

**Pull request merged**